chain/client/src/client_actor.rs

```diff
@@ -1088,8 +1096,12 @@ impl ClientActor {
                                 highest_height_peer(&self.network_info.highest_height_peers)
                             {
                                 if let Ok(header) = self.client.chain.get_block_header(&sync_hash) {
-                                    let prev_hash = header.prev_hash;
-                                    self.request_block_by_hash(prev_hash, peer_info.peer_info.id);
+                                    for hash in vec![header.prev_hash, header.hash].into_iter() {
```

Comment:
Why do you need to fetch `header.hash` at this point. AFAIU to sync state you only need to fetch `header.prev_hash` and `header.hash` will be requested while catching up. 

This still makes sense since we know we shortly will be requesting such block anyway, just want to be sure if it is a simple performance update, or if it is related somehow to the goal of the PR.

Reply:
See my PR description.